gtests/test_nopython_transform.cpp

```diff
@@ -0,0 +1,112 @@
+#include <modmesh/modmesh.hpp>
+#include <modmesh/transform/transform.hpp>
+#include <random>
+#include <gtest/gtest.h>
+
+#ifdef Py_PYTHON_H
+#error "Python.h should not be included."
+#endif
+
+template <typename T>
+class ParsevalTest : public ::testing::Test
+{
+protected:
+    const size_t VN = 1024;
+
+    modmesh::SimpleArray<modmesh::Complex<T>> signal{
+        modmesh::small_vector<size_t>{VN}, modmesh::Complex<T>(0.0, 0.0)};
+    modmesh::SimpleArray<modmesh::Complex<T>> out{
+        modmesh::small_vector<size_t>{VN}, modmesh::Complex<T>(0.0, 0.0)};
+
+    // Set up the test fixture: generate the signal once
+    void SetUp() override
+    {
+        std::mt19937 rng{std::random_device{}()};
+        std::uniform_real_distribution<T> val_dist{-1.0, 1.0};
+        for (unsigned int i = 0; i < VN; ++i)
+        {
+            T val = val_dist(rng);
+            signal[i] = modmesh::Complex<T>(val, 0.0);
+        }
+    }
+
+    // Function to verify energy conservation (Parseval's theorem)
+    void verify_parseval()
+    {
+        T psd_sig = 0.0, psd_out = 0.0;
+
+        for (unsigned int i = 0; i < VN; ++i)
+        {
+            psd_sig += signal[i].norm();
+            psd_out += out[i].norm();
+        }
+
+        // TODO: When FFT / DFT is using float, the accumulation error grows rapidly, we should fix it.
+        // Expect the total energy in the time and frequency domains to be equal
+        EXPECT_NEAR(psd_sig, psd_out, (std::is_same<T, float>::value ? (T)1e-2 : (T)1e-10));
```

Comment:
I left a TODO here to mention that the accumulation error grows rapidly when fft / dft are using `float`.
Also, relax the tolerance of `float` test cases, otherwise it may fail randomly.

Reply:
Please be explicit for how the error grows.  It is unclear what "grows rapidly" means.

Reply:
I have updated the comment to include the error ranges for `float` and `double`, describing the difference between the two data types.